mlflow/store/sqlalchemy_store.py

```diff
@@ -0,0 +1,200 @@
+import sqlalchemy
+from sqlalchemy import orm
+from sqlalchemy.exc import IntegrityError
+from mlflow.store.dbmodels import models
+from mlflow import entities
+from mlflow.entities import ViewType
+from mlflow.exceptions import MlflowException
+import mlflow.protos.databricks_pb2 as error_codes
+
+
+class SqlAlchemyStore(object):
+
+    def __init__(self, db_uri='sqlite://'):
+
+        self.engine = sqlalchemy.create_engine(db_uri)
+        models.Base.metadata.create_all(self.engine)
+        models.Base.metadata.bind = self.engine
+        db_session = orm.sessionmaker(bind=self.engine)
+        self.session = db_session()
+        print('Store started')
+
+    def _save_to_db(self, objs):
+        """
+        Store in db
+        """
+
+        if type(objs) is list:
+            self.session.add_all(objs)
+        else:
+            # single object
+            self.session.add(objs)
+
+        self.session.commit()
+
+    def list_experiments(self, view_type=ViewType.ACTIVE_ONLY):
+        mappings = {
+            ViewType.ACTIVE_ONLY: entities.Experiment.ACTIVE_LIFECYCLE,
+            ViewType.DELETED_ONLY: entities.Experiment.DELETED_LIFECYCLE
+        }
+
+        lifecycle_stage = mappings[view_type]
+        experiments = []
+        for exp in self.session.query(models.SqlExperiment).filter_by(
+                lifecycle_stage=lifecycle_stage):
+            experiments.append(exp.to_mlflow_entity())
+
+        return experiments
+
+    def create_experiment(self, name, artifact_location=None):
+        if name is None or name == '':
+            raise MlflowException('Invalid experiment name', error_codes.INVALID_PARAMETER_VALUE)
+        try:
+            experiment = models.SqlExperiment(
+                name=name,  lifecycle_stage=entities.Experiment.ACTIVE_LIFECYCLE,
+                artifact_location=artifact_location
+            )
+            self.session.add(experiment)
+            self.session.commit()
+        except IntegrityError:
+            self.session.rollback()
+            raise MlflowException('Experiment(name={}) already exists'.format(name),
+                                  error_codes.RESOURCE_ALREADY_EXISTS)
+
+        return experiment.to_mlflow_entity()
+
+    def get_experiment(self, experiment_id):
+        exp = self.session.query(models.SqlExperiment).filter_by(
+            experiment_id=experiment_id).first()
+
+        if not exp:
+            raise MlflowException('No Experiment with id={} exists'.format(experiment_id),
+                                  error_codes.RESOURCE_DOES_NOT_EXIST)
+
+        return exp.to_mlflow_entity()
+
+    def delete_experiment(self, experiment_id):
+        self.session.query(models.SqlExperiment).filter_by(
```

Comment:
Can this be implemented with a `is_delete` or `deleted_timestamp` flag in `experiments` table to be able to restore? `FileStore` has a mechanism to restore deleted experiments--this implementation needs to be compatible.